### PR TITLE
Issue #9921 automated tests for es snapshot

### PR DESCRIPTION
**Pull request by maurizo**

### CODE REVIEW

File: test/com/dotcms/content/elasticsearch/business/ESIndexAPITest.java [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+			esIndexAPI.clearIndex(indexName);
+			esIndexAPI.closeIndex(indexName);
+			//esIndexAPI.delete(indexName);
+			ZipFile file = new ZipFile(snapshot.getAbsolutePath());
+			String pathToRepo = Config.getStringProperty("es.path.repo","test-resources");
+			File tempDir = new File(pathToRepo);
+			boolean response = esIndexAPI.uploadSnapshot(file, tempDir.getAbsolutePath());
+			assertTrue(response);
+		} catch (DotStateException | IllegalArgumentException | IOException e) {
+			e.printStackTrace();
```
Comment (dsilvam) [inline test/com/dotcms/content/elasticsearch/business/ESIndexAPITest.java]: Same as above. Remove try-catch

File: test/com/dotcms/content/elasticsearch/business/ESIndexAPITest.java [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+		String indexName = getLiveIndex();
+		File snapshot = null;
+		try {
+			snapshot = esIndexAPI.createSnapshot(ESIndexAPI.BACKUP_REPOSITORY, "backup", indexName);
+			ZipFile file = new ZipFile(snapshot.getAbsolutePath());
+			String pathToRepo = Config.getStringProperty("es.path.repo","test-resources");
+			File tempDir = new File(pathToRepo);
+			esIndexAPI.uploadSnapshot(file, tempDir.getAbsolutePath());
+		} catch (DotStateException | IllegalArgumentException | IOException e) {
+			e.printStackTrace();
```
Comment (dsilvam) [inline test/com/dotcms/content/elasticsearch/business/ESIndexAPITest.java]: Same as above. Remove try-catch

File: test/com/dotcms/content/elasticsearch/business/ESIndexAPITest.java [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+
+	@AfterClass
+	public static void cleanUpTests(){
+		String pathToRepo = Config.getStringProperty("es.path.repo","test-resources");
+		File baseDirectory = new File(pathToRepo);
+		File toDelete = baseDirectory.getParentFile();
+		try {
+			FileUtil.deleteDir(toDelete.getAbsolutePath());
+		} catch (IOException e) {
+			e.printStackTrace();
```
Comment (dsilvam) [inline test/com/dotcms/content/elasticsearch/business/ESIndexAPITest.java]: Use Logger or remove try-catch and throw IOException

File: src/com/dotcms/content/elasticsearch/business/ESIndexHelper.java [hunk 4]
```diff
@@ -16,28 +24,61 @@
...
+//        private static final ESIndexHelper INSTANCE = new ESIndexHelper();
+//    }
+//    /**
+//     * Get the instance.
+//     * @return ESIndexHelper
+//     */
+//    public static ESIndexHelper getInstance() {
+//
+//        return ESIndexHelper.SingletonHolder.INSTANCE;
+//    } // getInstance.
```
Comment (dsilvam) [inline src/com/dotcms/content/elasticsearch/business/ESIndexHelper.java]: Can this commented-out block be removed?

**Review by dsilvam — CHANGES_REQUESTED** [review]

**Review by jgambarios — APPROVED** [review]

**Review by jdotcms — APPROVED** [review]

File: src/com/dotcms/content/elasticsearch/business/ESIndexHelper.java (hunk 4 above)
Comment (dsilvam) [inline src/com/dotcms/content/elasticsearch/business/ESIndexHelper.java]: what about this?

File: test/com/dotcms/content/elasticsearch/business/ESIndexAPITest.java (hunk 3 above)
Comment (dsilvam) [inline test/com/dotcms/content/elasticsearch/business/ESIndexAPITest.java]: this one is pending

**Review by dsilvam — CHANGES_REQUESTED** [review]

**Review by dsilvam — APPROVED** [review]

**Pull request merged**